# When building blue-alsa with more codecs,
# each codec must be specified to be used.
--- a/etc/init.d/bluetooth	2021-06-28 04:23:39.000000000 +0200
+++ b/etc/init.d/bluetooth	2024-12-20 14:47:01.932052590 +0100
@@ -15,12 +15,21 @@
 PROG3=/usr/bin/bluez_mibt_classical
 PROG5=/usr/bin/bluez_mibt_ble
 
+function bluealsa_get_codecs() {
+	CODECS=""
+	for NAME in `bluealsa --help  | grep -i -A5 "Available BT audio codecs:" | grep "a2dp-sink:" | awk '{$1=""; print $0}' | tr -d ','` ; do
+		CODECS="${CODECS} -c ${NAME}"
+	done
+}
+
 function bluealsa_start() {
+	bluealsa_get_codecs
+
 	procd_open_instance
 	if [ $DEBUG = "1" ]; then
-		procd_set_param command "$PROG1" -i hci0 -p a2dp-sink -S
+		procd_set_param command "$PROG1" -i hci0 -p a2dp-sink -S ${CODECS}
 	else
-		procd_set_param command "$PROG1" -i hci0 -p a2dp-sink
+		procd_set_param command "$PROG1" -i hci0 -p a2dp-sink ${CODECS}
 	fi
 	procd_append_param respawn 3600
 	procd_append_param respawn 5
